src/vcpkg/base/hash.cpp

```diff
@@ -557,36 +559,41 @@ namespace vcpkg::Hash
 
     ExpectedL<std::string> get_file_hash(const Filesystem& fs, const Path& path, Algorithm algo)
     {
+        Debug::println("Trying to hash ", path);
         std::error_code ec;
         auto file = fs.open_for_read(path, ec);
-        if (!ec)
+        if (ec)
         {
-            auto result = do_hash(algo, [&file, &ec](Hasher& hasher) {
-                constexpr std::size_t buffer_size = 1024 * 32;
-                char buffer[buffer_size];
-                do
+            return msg::format(msg::msgErrorMessage)
+                .append(msgHashFileFailureToRead, msg::path = path)
+                .append_raw(ec.message());
+        }
+
+        return do_hash<ExpectedL<std::string>>(algo, [&](Hasher& hasher) -> ExpectedL<std::string> {
+            constexpr std::size_t buffer_size = 1024 * 32;
+            char buffer[buffer_size];
+            do
+            {
+                const auto this_read = file.read(buffer, 1, buffer_size);
+                if (this_read != 0)
                 {
-                    const auto this_read = file.read(buffer, 1, buffer_size);
-                    if (this_read != 0)
-                    {
-                        hasher.add_bytes(buffer, buffer + this_read);
-                    }
-                    else if ((ec = file.error()))
-                    {
-                        return std::string();
-                    }
-                } while (!file.eof());
-                return hasher.get_hash();
-            });
+                    hasher.add_bytes(buffer, buffer + this_read);
+                }
+                else if ((ec = file.error()))
+                {
+                    return msg::format(msg::msgErrorMessage)
+                        .append(msgHashFileFailureToRead, msg::path = path)
+                        .append_raw(ec.message());
+                }
+            } while (!file.eof());
 
-            if (!ec)
+            auto result_hash = hasher.get_hash();
+            if (Debug::g_debugging)
```

Comment:
I don't think we need this if statement, `Debug::print` should already check if we are in debug mode.

Reply:
This was done to avoid the cost of the `format` but that's premature, yes.